tests/performance_meter/test_performance.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import asyncio
+import time
+import random
+import statistics
+import pytest
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+
+async def async_command(duthost, command):
+    return duthost.command(command)
+
+
+async def async_command_ignore_errors(duthost, command):
+    try:
+        return duthost.command(command, module_ignore_errors=True)
+    except Exception:
+        return
+
+
+# Defining an op.
+# An op is seperated into 2 parts by yield. The first part is setup
+# and happens before checking for success criteria, and the second
+# part is cleanup and happens after checking for success criteria
+# and the goal is to make sure that no change is leftover.
+# An op can be blocking or nonblocking, or combined, depending on
+# the need. For example, sometimes we want reboot to block until it
+# is successfully done before proceeding, or we want reboot to not
+# block at all to calculate how much time it takes to reboot.
+# If an op does not have yield in it, it will be treated as blocking.
+# Timing will only start after the first blocking part of operation
+# is over. The op should make sure op is started correctly and ended
+# correctly. If either part is unsuccessful, op should yeild False and
+# log the error, otherwise yielding True is expected.
+
+
+async def noop(duthost):
+    yield True
+
+
+async def bad_op(duthost):
+    yield False
+
+
+async def reboot_by_cmd(duthost):
+    command = asyncio.create_task(async_command_ignore_errors(duthost, "reboot"))
+    yield True
+    await command
+
+
+# Defining a success criteria and its stats.
+# A success criteria is a function defined in this module that
+# returns a function that returns True or False. It takes a duthost
+# and all variables defined in config that starts with the
+# name of said criteria, as keyword args. If we have "bgp_up",
+# then it will take "bgp_up_timeout", "bgp_up_delay", "bgp_up_foo",
+# etc as kwargs. A timeout is expected because we don't test to hang
+# forever. A delay is to not run the check for said time, default
+# to 0. Because each test run is separate, the function cannot
+# process results of all runs, so there could be a success criteria
+# stats function, named with a "_stats" suffix, taking the same
+# variables as its single run version, like "bgp_up_stats". It will
+# take all results that passed op precheck.
+
+
+def random_success_20_perc(duthost, **kwarg):
```

Comment:
This is a sample implementation, right?
If user use this one, their test will got ~20% passing rate